File: Javier_Programacion/infoBA.spring.-Solucion-Security/src/main/java/com/daw2/infoba/model/entity/Cliente.java
package com.daw2.infoba.model.entity;

import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import java.util.Date;
import java.util.List;

@Entity
//@Table(name="clientes",
//        uniqueConstraints= {@UniqueConstraint (columnNames= {"nombre", "apellido1"})})
@Table(name="clientes")
public class Cliente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true,nullable = false,length = 12)
    private String nif;
    @NotBlank
    private String razonSocial;
    @NotBlank
    @Column(nullable = false,length = 20)
    private String nombre;
    @NotBlank
    @Column(nullable = false,length = 20)
    private String apellido1;
    @Column(length = 20)
    private String apellido2;
    @Column(length = 15)
    private String telefono;
    @NotBlank
    @Email
    @Column(length = 75)
    private String email;
    @Column(name = "create_at", nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @Temporal(TemporalType.TIMESTAMP)
    private Date createAt;

    @OneToMany(mappedBy = "cliente")
    private List<Factura> facturas;


    @PrePersist
    public void init() {
        createAt = new Date();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNif() {
        return nif;
    }

    public void setNif(String nif) {
        this.nif = nif;
    }

    public String getRazonSocial() {
        return razonSocial;
    }

    public void setRazonSocial(String razonSocial) {
        this.razonSocial = razonSocial;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido1() {
        return apellido1;
    }

    public void setApellido1(String apellido1) {
        this.apellido1 = apellido1;
    }

    public String getApellido2() {
        return apellido2;
    }

    public void setApellido2(String apellido2) {
        this.apellido2 = apellido2;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Date getCreateAt() {
        return createAt;
    }

    public void setCreateAt(Date createAt) {
        this.createAt = createAt;
    }

    public List<Factura> getFacturas() {
        return facturas;
    }

    public void setFacturas(List<Factura> facturas) {
        this.facturas = facturas;
    }
}

File: erpdaw2/pojos/Cliente.php
<?php 
	class Cliente {
		private $id;
		private $idCliente;
		private $idUsuario;
		private $nombre;
		private $apellido1;
		private $apellido2;
		private $nif;
		private $numCta;
		private $comoNosConocio;
		private $activo;



	public function __construct($id, $idCliente, $idUsuario, $nombre, $apellido1, $apellido2, $nif, $numCta, $comoNosConocio, $activo)
	{
		$this->id = $id;
		$this->idCliente = $idCliente;
		$this->idUsuario = $idUsuario;
		$this->nombre = $nombre;
		$this->apellido1 = $apellido1;
		$this->apellido2 = $apellido2;
		$this->nif = $nif;
		$this->numCta = $numCta;
		$this->comoNosConocio = $comoNosConocio;
		$this->activo = $activo;
	}




	
    public function getId()
    {
        return $this->id;
    }

    
    public function setId($id)
    {
        $this->id = $id;

        return $this;
    }

    
    public function getIdCliente()
    {
        return $this->idCliente;
    }

    public function setIdCliente($idCliente)
    {
        $this->idCliente = $idCliente;

        return $this;
    }

    public function getIdUsuario()
    {
        return $this->idUsuario;
    }


    public function setIdUsuario($idUsuario)
    {
        $this->idUsuario = $idUsuario;

        return $this;
    }

  
    public function getNombre()
    {
        return $this->nombre;
    }


    public function setNombre($nombre)
    {
        $this->nombre = $nombre;

        return $this;
    }

    public function getApellido1()
    {
        return $this->apellido1;
    }


    public function setApellido1($apellido1)
    {
        $this->apellido1 = $apellido1;

        return $this;
    }

  
    public function getApellido2()
    {
        return $this->apellido2;
    }

 
    public function setApellido2($apellido2)
    {
        $this->apellido2 = $apellido2;

        return $this;
    }

    public function getNif()
    {
        return $this->nif;
    }


    public function setNif($nif)
    {
        $this->nif = $nif;

        return $this;
    }

   
    public function getNumCta()
    {
        return $this->numCta;
    }

    public function setNumCta($numCta)
    {
        $this->numCta = $numCta;

        return $this;
    }

   
    public function getComoNosConocio()
    {
        return $this->comoNosConocio;
    }

 
    public function setComoNosConocio($comoNosConocio)
    {
        $this->comoNosConocio = $comoNosConocio;

        return $this;
    }

 
    public function getActivo()
    {
        return $this->activo;
    }

   
    public function setActivo($activo)
    {
        $this->activo = $activo;

        return $this;
    }
}




 ?>